sunpy/database/tests/test_database.py

```diff
@@ -785,6 +787,40 @@ def test_fetch(database, download_query, tmpdir):
 
 
 @pytest.mark.online
+def test_fetch_separate_filenames():
+    # Setup
+    db = Database('sqlite:///')
+
+    download_query = [
+        vso.attrs.Time('2012-08-05', '2012-08-05 00:00:05'),
+        vso.attrs.Instrument('AIA')
+    ]
+
+    tmp_test_dir = os.path.join(
+        sunpy.config.get('downloads', 'download_dir'),
+        'tmp_test_dir/'
+    )
+
+    if not os.path.isdir(tmp_test_dir):
+        os.mkdir(tmp_test_dir)
+
+    path = tmp_test_dir + '{file}.fits'
+
+    db.fetch(*download_query, path=path)
+    
+    # Test
+    assert len(db) == 2
+    
+    dir_contents = os.listdir(tmp_test_dir)
+    assert dir_contents[-2] == 'aia_lev1_335a_2012_08_05t00_00_02_62z_image_lev1_fits.fits'
+    assert dir_contents[-1] == 'aia_lev1_94a_2012_08_05t00_00_01_12z_image_lev1_fits.fits'
+    assert dir_contents[-1].split('.')[0] != dir_contents[-2].split('.')[1]
```

Comment:
I looked at the errors in travis.....the test I added is passing on travis too.....but the ones which are failing are the heuristic tests in test_mapbase.py.......I will open an issue for that and then fix them as ones in test_callisto.py ( #1332)....using asset_allclose....